internal/driver/webui.go

```diff
@@ -179,11 +179,21 @@ func defaultWebServer(args *plugin.HTTPServerArgs) error {
 	// https://github.com/google/pprof/pull/348
 	mux := http.NewServeMux()
 	mux.Handle("/ui/", http.StripPrefix("/ui", handler))
-	mux.Handle("/", http.RedirectHandler("/ui/", http.StatusTemporaryRedirect))
+	mux.Handle("/", redirectWithQuery("/ui"))
 	s := &http.Server{Handler: mux}
 	return s.Serve(ln)
 }
 
+func redirectWithQuery(url string) http.HandlerFunc {
+	urlWithQuery := url
+	return func(w http.ResponseWriter, r *http.Request) {
+		if queryStr := r.URL.Query().Encode(); queryStr != "" {
```

Comment:
I feel this could be just

```
mux.Handle("/", func(w http.ResponseWriter, r *http.Request) {
  http.Redirect(w, r, url.URL{Path: "/ui/", RawQuery: r.URL.RawQuery}.String(), http.StatusTemporaryRedirect)
}
```

?

Reply:
I would inline the function to be a lambda to avoid mental load "oh but where redirectWithQuery is used? ah, just this one place" which is obvious with the inlined lambda. But it's fine as is.

Reply:
I think it would have to be wrapped in http.HandlerFunc(...) which I didn't think was very readable, but I can make the change if you feel strongly

Reply:
I hoped it would implicitly cast, but I may be wrong. Not feeling strongly. Waiting for CI to pass before I merge.

Reply:
Thanks for the fix BTW!

Reply:
yeah no problem, thanks for the useful tool, I use it almost everyday